File: backend/internal/domain/entities/edge.go
package entities

type Edge struct {
	ID     string  `json:"edgeId" bson:"edgeId"` // Уникальный ID ребра
	Source string  `json:"source" bson:"source"` // Источник (ID узла)
	Target string  `json:"target" bson:"target"` // Цель (ID узла)
	Label  string  `json:"label" bson:"label"`   // Метка ребра
	Color  string  `json:"color" bson:"color"`   // Цвет ребра
	Size   float64 `json:"size" bson:"size"`     // Размер ребра (= Weight)
	Type   string  `json:"type" bson:"type"`     // Тип связи: дружба/работа/родство/знакомство/бизнес
	Weight int     `json:"weight" bson:"weight"` // Сила связи 1–5
}

File: web-app/src/types/edge.ts

export interface Edge {
  edgeId: string;
  source: string;
  target: string;
  label?: string;
  color?: string;
  size?: number;
  type?: string;
  weight?: number;
}

export interface AddContactEdge {
  source: string,
  label: string,
  color: string,
  size: number,
  type: string,
  weight: number,
}

export const EDGE_TYPES = ['дружба', 'работа', 'родство', 'знакомство', 'бизнес'] as const;
export type EdgeType = typeof EDGE_TYPES[number];